src/hera/workflows/io/v2.py

```diff
@@ -0,0 +1,105 @@
+"""Pydantic V2 input/output models for the Hera runner.
+
+RunnerInput/Output are only defined in this file if Pydantic v2 is installed.
+"""
+from collections import ChainMap
+from typing import Any, List, Union
+
+from hera.shared.serialization import serialize
+from hera.workflows.artifact import Artifact
+from hera.workflows.parameter import Parameter
+
+try:
+    from inspect import get_annotations  # type: ignore
+except ImportError:
+    from hera.workflows._inspect import get_annotations  # type: ignore
+
+try:
+    from typing import Annotated, get_args, get_origin  # type: ignore
+except ImportError:
+    from typing_extensions import Annotated, get_args, get_origin  # type: ignore
+
+from importlib.util import find_spec
+
+if find_spec("pydantic.v1"):
```

Comment:
@samj1912 does this make sense to you? Any other ways we can/should do this? Also on file naming - I think  it makes it clear which version clients should use, and it makes sense that the `v2.py` file only defines `RunnerInput`/`RunnerOutput` if pydantic v2 is installed.